lua/multicursors/utils.lua

```diff
@@ -493,4 +493,75 @@ M.exit = function()
     vim.b.MultiCursorSubLayer = nil
 end
 
+-- Generates hints based on the configuration and input parameters.
+---@param config Config configuration.
+---@param heads Head[]
+---@param mode string indicating the mode.
+---@return string hints as a string.
+M.generate_hints = function(config, heads, mode)
+    if config.generate_hints[mode] == false then
+        return 'MultiCursor ' .. mode .. ' mode'
+    elseif type(config.generate_hints[mode]) == 'string' then
+        return config.generate_hints[mode]
+    end
+
+    local num_items = #heads
+    local num_items_to_sort = num_items
+    local last_five_items = {}
+
+    -- Keep the last 5 items in the same pos.
+    -- Esc -> i -> c -> a -> e
+    if mode == 'normal' then
+        num_items_to_sort = math.max(num_items - 5, 0)
+        for i = num_items_to_sort + 1, num_items do
+            table.insert(last_five_items, heads[i])
+        end
+    end
+
+    ---@type Head[]
+    local items_to_sort = {}
+    for i = 1, num_items_to_sort do
+        table.insert(items_to_sort, heads[i])
+    end
+
+    table.sort(items_to_sort, function(a, b)
+        -- Custom sorting logic here (alphabetical sorting with special characters at the end)
+        local is_special_a = not string.match(a[1], '[%a%d]')
+        local is_special_b = not string.match(b[1], '[%a%d]')
```

Comment:
I don't understand these lines what are those patterns?

Reply:
Custom sorting logic here (alphabetical sorting with special characters at the end)